packages/relay/src/services/PostService.ts

```diff
@@ -225,29 +228,47 @@ export class PostService {
 
         // post content
         const cid = await IpfsHelper.createIpfsContent(helia, content)
-        const txnHash = await TransactionManager.callContract('post', [
+
+        const appContract = new ethers.Contract(APP_ADDRESS, APP_ABI)
+        const calldata = appContract.interface.encodeFunctionData('post', [
             epochKeyProof.publicSignals,
             epochKeyProof.proof,
             content,
         ])
 
+        const { txHash, logs } = await TransactionManager.executeTransaction(
+            appContract,
+            APP_ADDRESS,
+            calldata
+        )
+
+        // FIXME should be latest post id after contract changes
+        const postId = logs
+            .filter(
+                (log): log is ethers.utils.LogDescription =>
+                    log !== null && log.name === 'Post'
+            )
+            .map((log) => log.args.postId)
+            .find((postId) => postId !== undefined)
+            .toString()
+
         const epoch = Number(epochKeyProof.epoch)
         const epochKey = epochKeyProof.epochKey.toString()
 
         // after post data stored in DB, should add 1 to epoch key counter
         await ActionCountManager.addActionCount(db, epochKey, epoch, (txDB) => {
```

Comment:
Maybe we should move add action counter to Synchronizer also, since these actions are atomic

Reply:
Thank you for the suggestions. Would it be better that we keep the action count adjustment in Services? Since it will be more consistent, cause we won't be able to identify whether this transaction is using queueTx or executeTx. So readers won't be confused that some handle events have adjusted action count and some don't.

Reply:
Understood, but no matter queueTx or executeTx both should go through eventHandler, if we could put counter logic with insert, it will be easily to extend and develop in the future (could easily refer to other event method). Also, for consistency, if we keep all logic in eventHandler, I believe there is no different case for User Experience, for example, some services is using queueTx with counter, user will see counter increase immediately, but the others will be opposite.